accelerator/core/core.c

```diff
@@ -780,17 +795,27 @@ status_t ta_fetch_txn_with_uuid(const ta_cache_t* const cache, const char* const
   if (exist) {
     res->status = SENT;
 
-    ret = cache_get(uuid, (char*)txn_flex_trits);
+    int len = 0;
+    ret = cache_list_size(uuid, &len);
     if (ret) {
       ta_log_error("%s\n", ta_error_to_string(ret));
       goto done;
     }
 
-    res->txn = transaction_deserialize(txn_flex_trits, false);
-    if (res->txn == NULL) {
-      ret = SC_CORE_OOM;
-      ta_log_error("%s\n", ta_error_to_string(ret));
-      goto done;
+    for (int i = 0; i < len; ++i) {
+      flex_trit_t txn_flex_trits[NUM_FLEX_TRITS_SERIALIZED_TRANSACTION + 1];
+      ret = cache_list_at(uuid, i, NUM_FLEX_TRITS_SERIALIZED_TRANSACTION, (char*)txn_flex_trits);
+      if (ret) {
+        ta_log_error("%s\n", ta_error_to_string(ret));
+        goto done;
+      }
+      iota_transaction_t* txn = transaction_deserialize(txn_flex_trits, false);
+
+      if (bundle_transactions_add(res->bundle, txn) != RC_OK) {
+        ta_log_error("%s\n", "Failed to add transaction to bundle.");
```

Comment:
What is the `ret` value in this case?

Reply:
The case happened only `res->bundle`, or `txn` equals to NULL.